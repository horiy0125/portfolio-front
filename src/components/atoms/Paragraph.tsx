import React from 'react';
import styled from 'styled-components';
import { classes } from '@horri1520/hori-base-elements/lib/utils/classes';

type Props = {
  children?: React.ReactNode;
  className?: string;
};

const Root = styled.p`
  line-height: 1.8;
`;

const Paragraph: React.VFC<Props> = ({ children, className }) => {
  return <Root className={classes(className)}>{children}</Root>;
};

export default Paragraph;
