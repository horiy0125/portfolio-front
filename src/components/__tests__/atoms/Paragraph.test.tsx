import React from 'react';
import renderer from 'react-test-renderer';
import Paragraph from '../../atoms/Paragraph';

describe('components/atoms/Paragraph', () => {
  describe('if the component has string children', () => {
    it('render children as it is', () => {
      const component = renderer.create(<Paragraph>p</Paragraph>);
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('if the component has React.ReactNode as children', () => {
    it('render React.ReactNode as children', () => {
      const component = renderer.create(
        <Paragraph>
          <a href="https://example.com" id="link">
            Link
          </a>
        </Paragraph>,
      );
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('if the component has no children', () => {
    it('render only myself', () => {
      const component = renderer.create(<Paragraph></Paragraph>);
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});

export {};
