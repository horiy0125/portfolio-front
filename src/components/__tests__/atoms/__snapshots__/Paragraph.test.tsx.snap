// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/Paragraph if the component has React.ReactNode as children render React.ReactNode as children 1`] = `
<p
  className="sc-bdnxRM gVrMcJ"
>
  <a
    href="https://example.com"
    id="link"
  >
    Link
  </a>
</p>
`;

exports[`components/atoms/Paragraph if the component has no children render only myself 1`] = `
<p
  className="sc-bdnxRM gVrMcJ"
/>
`;

exports[`components/atoms/Paragraph if the component has string children render children as it is 1`] = `
<p
  className="sc-bdnxRM gVrMcJ"
>
  p
</p>
`;
